solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {FungibleTokenRouter} from "./FungibleTokenRouter.sol";
+import {ValueTransferBridge} from "./ValueTransferBridge.sol";
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+abstract contract MovableCollateralRouter is
+    FungibleTokenRouter,
+    AccessControl
+{
+    constructor(
+        uint256 _scale,
+        address _mailbox
+    ) FungibleTokenRouter(_scale, _mailbox) {
+        _grantRole(DEFAULT_ADMIN_ROLE, owner());
+    }
+
+    bytes32 public constant REBALANCER_ROLE = keccak256("REBALANCER_ROLE");
+
+    mapping(uint32 destinationDomain => bytes32 recipient)
+        public allowedDestinations;
+    mapping(uint32 destinationDomain => mapping(ValueTransferBridge bridge => bool isValidBridge))
+        public allowedBridges;
+
+    event CollateralMoved(
+        uint32 indexed domain,
+        bytes32 recipient,
+        uint256 amount,
+        address indexed rebalancer
+    );
+
+    error BadDestination(address rebalancer, uint32 domain);
+    error BadBridge(address rebalancer, ValueTransferBridge bridge);
+
+    function moveCollateral(
+        uint32 domain,
+        bytes32 recipient,
+        uint256 amount
+    ) external payable onlyRole(REBALANCER_ROLE) {
+        _moveCollateral(
+            domain,
+            recipient,
+            amount,
+            ValueTransferBridge(address(this))
```

Comment:
interesting

Reply:
Yeah, but I'll remove it since we could always just call `addBridge` on the router and pass in its own address.